src/components/Articles.jsx

```diff
@@ -7,21 +7,28 @@ const Articles = () =>{
     const search = window.location.search;
     
     const [articlesList, setArticlesList] = useState([])
+    const [isLoading, setIsLoading] = useState(false)
   
     useEffect(()=> {
+      setIsLoading(true)
        getArticles().then((response)=> { if(search) {
         const topicName = search.slice(7)
         const filteredArticles = response.articles.filter((article) => {
             return article.topic === topicName
         })
```

Comment:
You need to filter in the api request. You set up the backend to be able to do this.